tools/upgrade.sh

```diff
@@ -69,10 +69,16 @@ then
   printf "${BLUE}${BOLD}%s ${UNDER}%s${RESET}\n" "Get your Oh My Zsh swag at:" "https://shop.planetargon.com/collections/oh-my-zsh"
 else
   printf "${RED}%s${RESET}\n" 'There was an error updating. Try again later?'
+
+  # Prepare to return an error exit code
+  status=1
 fi
 
 # Unset git-config values set just for the upgrade
 case "$resetAutoStash" in
   "") git config --unset rebase.autoStash ;;
   *) git config rebase.autoStash "$resetAutoStash" ;;
 esac
+
+# Exit with `1` if the update failed
+exit $status && unset status
```

Comment:
The `unset` command will never fire after running `exit` (try it), and since this script is called with sh anyways it's not necessary to "clean up" afterwards.